storage/replica_command.go

```diff
@@ -536,21 +540,43 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, rangeCount,
 			args.Timestamp, consistent, args.Txn)
 	} else {
-		// Use MVCCScan to get first the first range. There are two cases:
-		// 1. args.Key is not an endpoint of the range and
-		// 2. The args.Key is the start/end key of the range.
+		// Use MVCCScan to get the first range. There are three cases:
+		// 1. args.Key is not an endpoint of the range, and
+		// 2a. The args.Key is the start/end key of the range.
+		// 2b. Even worse, the body of args.Key is proto.KeyMax.
 		// In the first case, we need use the MVCCScan() to get the first range
-		// descriptor, because the semantic of ReverseScan can't do the work.
-		// For example: If we have ranges ["a","c") and ["c","f") and the
-		// reverse scan request's key range is ["b","d"), then "d".Next() is
-		// less than "f", and so the meta row {"f"->["c", "f")} would be
-		// ignored by MVCCReverseScan. In the second case, the range descriptor
-		// received by MVCCScan will be filtered before results are returned.
-		startKey, endKey := keys.MetaScanBounds(args.Key)
-		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, 1,
-			args.Timestamp, consistent, args.Txn)
-		if err != nil {
-			return reply, nil, err
+		// descriptor because ReverseScan can't do the work: If we have ranges
+		// ["a","c") and ["c","f"), and the reverse scan request's key range is
+		// ["b","d"), then "d".Next() is less than "f", and so the meta row
+		// {"f"->["c", "f")} would be ignored by MVCCReverseScan. In case 2a,
+		// the range descriptor received by MVCCScan will be filtered before
+		// results are returned: With ranges ["c", "f") and ["f","z"), reverse
+		// scan on ["d", "f") receives the descriptor {"z"->["f","z")}, which
+		// is discarded below since it's not being asked for. Finally, in case
+		// 2b, we don't even attempt the forward scan because it's neither
+		// defined nor required.
+		// TODO(tschottdorf): unit testing coverage.
+		if args.Key[len(keys.Meta2Prefix):].Less(proto.KeyMax) {
```

Comment:
Should we also change `MetaReverseScanBounds()` to handle `keyMax` correctly (assuming we cannot call `keyMax.Next()`)?